Svc/GenericHub/GenericHub.fpp

```diff
@@ -1,54 +1,187 @@
+    #    genericHub.bufferInReturn[1] -> bufferProducer1.bufferIn
+    # ----------------------------------------------------------------------
+
+    @ Port for sending events to the hub
+    sync input port eventIn: Fw.Log
+
+    @ Port for sending telemetry to the hub
+    sync input port tlmIn: Fw.Tlm
 
-    @ Telemetry input port for mimicking an event processor
-    sync input port TlmRecv: Fw.Tlm
+    @ Ports for sending serial data to the hub
+    @ You can connect any typed output ports to these input ports, so
+    @ long as the data carried by the ports is serialized by value.
+    @ Do not connect ports that emit Fw.Buffer objects, because these objects
+    @ store pointers to data that is not serialized across the port
+    @ interface. To connect output ports that emit buffers, use
+    @ buffersIn below.
+    sync input port serialIn: [GenericHubCfg.NumSerialInputPorts] serial
 
-    @ Event input port for mimicking an event processor
-    sync input port LogRecv: Fw.Log
+    @ Ports for sending buffer data to the hub
+    @ Output ports connected to these ports must emit Fw.Buffer objects.
+    @ On invocation, each of these ports allocates a new buffer B, copies the
+    @ data from the incoming buffer to B, and returns the incoming
+    @ buffer to the sender for deallocation.
+    @
+    sync input port bufferIn: [GenericHubCfg.NumBufferInputPorts] Fw.BufferSend
 
-    @ Cross-hub event output port
-    output port LogSend:  Fw.Log
+    @ Ports for returning buffers arriving on buffersIn
+    output port bufferInReturn: [GenericHubCfg.NumBufferInputPorts] Fw.BufferSend
 
-    @ Cross-hub telemetry output port
-    output port TlmSend: Fw.Tlm
+    @ bufferIn and bufferInReturn ports must match
+    match bufferIn with bufferInReturn
 
     # ----------------------------------------------------------------------
-    # General Ports
+    # Ports for sending data from the hub to a buffer driver
+    # ----------------------------------------------------------------------
+    # These ports establish the "send" interface from the hub to a buffer driver.
+    #
+    # Sample connections:
+    #
+    #    genericHub.allocate -> bufferManager.bufferGetCallee
+    #    genericHub.toBufferDriver -> bufferDriver.bufferIn
+    #    bufferDriver.bufferInReturn -> genericHub.toBufferDriverReturn
+    #    genericHub.deallocate -> bufferManager.bufferSendIn
     # ----------------------------------------------------------------------
 
-    @ Input array of generic ports to shuttle to the other side of the hub connection
-    sync input port portIn: [GenericHubInputPorts] serial
+    @ This interface provides ports allocate and deallocate
+    import Fw.BufferAllocation
 
-    @ Output array of generic ports shuttled from the other side of the hub connection
-    output port portOut: [GenericHubOutputPorts] serial
+    @ This interface provides ports toBufferDriver and toBufferDriverReturn
+    import Drv.PassiveBufferDriverClientSend
```

Comment:
Why not just use the `PassiveBufferDriverClient` interface as it represent both Send/Recv halves that we are importing here?